src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.cs

```diff
@@ -42,29 +49,213 @@ public partial class HybridWebViewHandler : IHybridWebViewHandler
-            : base(mapper ?? Mapper, commandMapper ?? CommandMapper)
-        {
-        }
 
-        IHybridWebView IHybridWebViewHandler.VirtualView => VirtualView;
 
-        PlatformView IHybridWebViewHandler.PlatformView => PlatformView;
+		/// <summary>
+		/// Handler for when the an Async JavaScript task has completed and needs to notify .NET.
+		/// </summary>
+		private void AsyncTaskCompleted(string taskId, string result)
+		{
+			// Look for the callback in the list of pending callbacks
+			if (!string.IsNullOrEmpty(taskId) && _asyncTaskCallbacks.TryGetValue(taskId, out var callback))
+			{
+				// Get the callback and remove it from the list
+				callback.SetResult(result);
+
+				// Remove the callback
+				_asyncTaskCallbacks.TryRemove(taskId, out var _);
+			}
+		}
+
+		void MessageReceived(string rawMessage)
+		{
+			if (string.IsNullOrEmpty(rawMessage))
+			{
+				throw new ArgumentException($"The raw message cannot be null or empty.", nameof(rawMessage));
+			}
+#if !NETSTANDARD2_0
+			var indexOfPipe = rawMessage.IndexOf('|', StringComparison.Ordinal);
+#else
+			var indexOfPipe = rawMessage.IndexOf("|", StringComparison.Ordinal);
+#endif
+			if (indexOfPipe == -1)
+			{
+				throw new ArgumentException($"The raw message must contain a pipe character ('|').", nameof(rawMessage));
+			}
+
+			var messageType = rawMessage.Substring(0, indexOfPipe);
+			var messageContent = rawMessage.Substring(indexOfPipe + 1);
+
+			switch (messageType)
+			{
+				case "InvokeMethodCompleted":
+					{
+						var sections = messageContent.Split('|');
+						var taskId = sections[0];
+						var result = sections[1];
+						AsyncTaskCompleted(taskId, result);
+					}
+					break;
+				case "RawMessage":
+					VirtualView?.RawMessageReceived(messageContent);
+					break;
+				default:
+					throw new ArgumentException($"The message type '{messageType}' is not recognized.", nameof(rawMessage));
+			}
+		}
+
+#if PLATFORM && !TIZEN
+		public static async void MapEvaluateJavaScriptAsync(IHybridWebViewHandler handler, IHybridWebView hybridWebView, object? arg)
+		{
+			if (arg is not EvaluateJavaScriptAsyncRequest request ||
+				handler.PlatformView is not MauiHybridWebView hybridPlatformWebView)
+			{
+				return;
+			}
+
+			if (handler.PlatformView is null)
+			{
+				request.SetCanceled();
+				return;
+			}
+
+			var script = request.Script;
+			// Make all the platforms mimic Android's implementation, which is by far the most complete.
+			if (DeviceInfo.Platform != DevicePlatform.Android)
+			{
+				script = EscapeJsString(script);
+
+				if (DeviceInfo.Platform != DevicePlatform.WinUI)
+				{
+					// Use JSON.stringify() method to converts a JavaScript value to a JSON string
+					script = "try{JSON.stringify(eval('" + script + "'))}catch(e){'null'};";
+				}
+				else
+				{
+					script = "try{eval('" + script + "')}catch(e){'null'};";
+				}
+			}
```

Comment:
This needs to use the `OperatingSystem.IsPlatformName()` instead of essentials.

Reply:
Fixed